File: Team5/Sonali_Priya/Pivot Index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        vector<int> left_sum(n, 0);
        vector<int> right_sum(n, 0);
        for(int i=1; i<n; i++){
            left_sum[i] = left_sum[i-1]+nums[i-1];
        }
        for(int i=n-2; i>=0; i--){
            right_sum[i] = right_sum[i+1]+nums[i+1];
        }
        
        for(int i=0; i<n; i++){
            if(right_sum[i] == left_sum[i])
                return i;
        }
        return -1;
    }
};

File: Team1/RiyaSingh/pivotIndex.java
class Solution {
    public int pivotIndex(int[] nums) {
        int lsum = 0, rsum = 0;
        for(int i = 0; i<nums.length; i++){
            for(int j = 0; j< i; j++){
                lsum += nums[j];
            }
            for(int j=i+1; j<nums.length; j++){
                rsum += nums[j];
            }
            if(lsum == rsum)
                //System.out.println(nums[i]);
                return i;
            lsum = 0;
            rsum = 0;
        }
        return -1;   
    }
    
}
